infra/discovery/src/listener/listener.js

```diff
@@ -206,8 +213,8 @@ async function main() {
      * event-cache
      */
     processedToBlock = Math.min(
-      contractsContext.marketplace.eventCache.lastQueriedBlock,
-      contractsContext.identityEvents.eventCache.lastQueriedBlock
+      contractsContext.marketplace.eventCache.lastEventBlock,
```

Comment:
Marketplace and Identity events happen at difference pace.
So for example it's possible we would see a very few identity events but lots of marketplace events.
Which would then cause marketplace events to get reprocessed many times because processedToBlock would only get advances when we see an identity event...
The listener only provides at least once guarantees but still I wonder if there is something we could do to avoid this reprocessing ?

Reply:
I went this direction for safety.  I can surely change this to max, if you think that's the better call.  The only other option I suppose would be refactoring the listener entirely to separate the two.

Reply:
We could indeed refactor the listener to have 1 loop per contract type and keep separate block cursors for each loop... It's a slightly bigger change but wouldn't be super hard I think...
I'd say either we do that or use Math.max. Your call. What do you think ?

Reply:
Refactored.  Uses individual cursors now.